### PR TITLE
Implemented Amplitude SDK type definition.

### PR DESCRIPTION
**Pull request by asido**
case 1. Add a new type definition.
- [ ] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [ ] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

**Comment by vvakame** [comment]
please rename `amplitude/amplitude.d.ts` to `amplitude-js/amplitude-js.d.ts`
https://github.com/amplitude/Amplitude-Javascript/blob/master/package.json#L2

File: amplitude-js/amplitude-js.d.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// Type definitions for Amplitude SDK 2.12.1
+// Project: https://github.com/amplitude/Amplitude-Javascript
+// Definitions by: Arvydas Sidorenko <https://github.com/Asido>
+// Definitions: https://github.com/Asido/DefinitelyTyped
+
+declare module amplitude {
+	interface Config {
+		batchEvents?: boolean;
```
Comment (mhegazy) [inline amplitude-js/amplitude-js.d.ts]: can you limit the indentation to 4 chars.

File: amplitude-js/amplitude-js.d.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	export function setGroup(groupType: string, groupName: string | string[]): void;
+
+	export function logEvent(event: string): void;
+	export function logEvent(event: string, data: Object): void;
+	export function logEvent(event: string, data: Object, callback: (httpCode: number, response: any) => void): void;
+
+	export var options: Config;
+}
+
+//declare var amplitude: AmplitudeStatic;
```
Comment (mhegazy) [inline amplitude-js/amplitude-js.d.ts]: remove the comment. 

also according to the docs, amplitude can be used as a module: https://github.com/amplitude/Amplitude-Javascript#loading-with-requirejs

so we should have:
```ts
declare module "amplitude" {
    export = amplitude;
}
```

**Comment by Asido** [comment]
> I do not see any declarations for ```getInstance```

It's because the next day when I opened this issue they released another major version v2 -> v3, which isn't backwards compatible and introduced ```getInstance()```.

How do you typically handle such cases here at DefinitelyTyped? Port to v3 and drop support for v2?

**Comment by jramsay** [comment]
@Asido : sorry for the delay! It looks like we missed your question above. As long as the header is correct, which it is in this case as it targets Amplitude SDK 2.12.1, you can check in the older version.  If you want you can write a d.ts for the new version instead.  If you needed to support both, the latest should be in the root and then place the older version under /amplitude-js/v2.  
In either case it looks like there are a few minor items above that need to be addressed before this PR can be merged - if you can make these changes I'll merge in the request. (ex: indentation, declare module, remove commented out code)

**Comment by minestarks** [comment]
@Asido thanks for the PR. Do you intend to make the suggested revisions? I'll go ahead and merge once you address the minor things called out above. Thanks!
>  (ex: indentation, declare module, remove commented out code)

**Comment by paulvanbrenk** [comment]
@Asido  gentle ping.

**Comment by Asido** [comment]
I am sorry for the delay. You'll receive another PR with changes soon.

File: amplitude-js/amplitude-js.d.ts (hunk 2 above)
Comment (Asido) [inline amplitude-js/amplitude-js.d.ts]: This won't work with TypeScript if ES6 output is set:
``` TypeScript
declare module "amplitude" {
    export = amplitude;
}
```
What about using this instead?:
``` TypeScript
declare module "amplitude" {
    export default amplitude;
}
```

**Comment by rbuckton** [comment]
@Asido: on July 20th, @jramsay said:
>  If you needed to support both, the latest should be in the root and then place the older version under /amplitude-js/v2.

Unless you plan to also include the v3 typings, there is currently no need to move the .d.ts underneath v2. It is fine to leave it in the root for now.

**Pull request merged**